src/kernels/kernelFinder.ts

```diff
@@ -22,9 +22,27 @@ export class KernelFinder implements IKernelFinder {
 
     constructor(@inject(IDisposableRegistry) private readonly disposables: IDisposableRegistry) {}
 
-    public registerKernelFinder(finder: IContributedKernelFinder) {
+    public registerKernelFinder(finder: IContributedKernelFinder): IDisposable {
         this._finders.push(finder);
-        this.disposables.push(finder.onDidChangeKernels(() => this._onDidChangeKernels.fire()));
+        const onDidChangeDisposable = finder.onDidChangeKernels(() => this._onDidChangeKernels.fire());
```

Comment:
Some small changes to the main KernelFinder. Notify that kernels changed if a new finder is registered, and return a disposable on registration so that Finders can clean themselves out of the list as they dispose.